### DISCUSSION TITLE
How do I apply token filter on all routes in an API except on the default route

### DISCUSSION THREAD

**Discussion by Elvis254** [codeigniter4/shield#1053] [Q&A] [answered]
### PHP Version

8.2.0

### CodeIgniter4 Version

4.4.5

### Shield Version

1.0.1

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

apache

### Database

MariaDB 10.2

### Did you customize Shield?

I customized the `UserModel` class to include a function to generate a full name of a user from their first and last name before insert and before update. I also customized the `User` entity class to include two functions to set the first and last name of a user after cleaning the string input and performing formatting on it like capitalization.

### What happened?

I applied `TokenAuth` filter on all routes but I don't know how not to apply it on the default route which serves as the home page of the API. Here is the default route code am using: `$routes->get('/', 'HomeController::indexPage');` How do I write it in the except array collection of routes in the `$globals` filter configuration section?

### Steps to Reproduce

Find a way to make `TokenAuth` filter to be applied to all pages except the ones indicated in the except array collection of routes in `$globals` filter configuration section including the default route of the API.

### Expected Output

The `TokenAuth` filter should not be applied on the default route as indicated in the except array collection of routes in ```$globals``` filter configuration section.

**Comment by datamweb** [comment, marked as answer]
Hi,
To be honest, I always prefer to use [applying-filters](https://codeigniter.com/user_guide/incoming/routing.html#applying-filters) instead of using [$globals](https://codeigniter.com/user_guide/incoming/filters.html#id9).

In any case, you must proceed as [except-for-a-few-uris](https://codeigniter.com/user_guide/incoming/filters.html#except-for-a-few-uris).

**Reply by Elvis254** [reply]
Thanks I have done that and it works.
